### PR TITLE
Migrate Travis job to the Circle CI to unify CI pipelines used

### PR DESCRIPTION
**Pull request by pschoffer**
## Description
In order to unify the CI pipelines we use we decided to move the only Travis one to the rest in circleCI.

**NOTE**

We might need to clean up config in here: https://travis-ci.org/Automattic/vip-go-mu-plugins/settings after merge.

### No production code change

This changelog entry can be ignored.

## Checklist

Please make sure the items below have been covered before requesting a review:

- [ ] This change works and has been tested locally (or has an appropriate fallback).
- [ ] This change works and has been tested on a Go sandbox.
- [ ] This change has relevant unit tests (if applicable).
- [ ] This change has relevant documentation additions / updates (if applicable).
- [ ] I've created a changelog description that aligns with the provided examples.

## Steps to Test

Check build pipelines

### CODE REVIEW

**Comment by pauarge** [comment]
We could also use this PR to remove all references to Travis (e.g. `travis.yml`).

**Comment by pschoffer** [comment]
Oh yeah it is far from gone... it is just that turns out you can't run the build pipeline locally if they need secret ssh keys... so I created the PR already so I get the circleCI runs.

File: ci/deploy.sh [hunk 1]
```diff
@@ -1,50 +1,26 @@
...
-chmod 600 ~/.ssh/id_rsa
+cp ./ci/known_hosts ~/.ssh/known_hosts
 
-# Restore script echoing now we've done the private things
-set -x
-
-cp ${TRAVIS_BUILD_DIR}/ci/known_hosts ~/.ssh/known_hosts
-
-git clone git@github.com:Automattic/vip-go-mu-plugins-built.git /tmp/target
+git clone  git@github.com:Automattic/vip-go-mu-plugins-built.git /tmp/target
```
Comment (pauarge) [inline ci/deploy.sh:17]: Super small nit-pick: there's a space added between `clone` and `git@` that's not necessary.

File: ci/generate-docs.sh [hunk 2]
```diff
@@ -1,51 +1,19 @@
...
+cp ./ci/known_hosts ~/.ssh/known_hosts
 
 # Clone the repo at the `gh-pages` branch, getting only the commits for that branch
-git clone --branch gh-pages --single-branch "git@github.com:${TRAVIS_REPO_SLUG}.git" ${VIP_DOCS_DIR}
+git clone --branch gh-pages --single-branch "git@github.com:Automattic/vip-go-mu-plugins.git" ${VIP_DOCS_DIR}
 
-# Composer runs faster without Xdebug, and we don't need Xdebug any more
-phpenv config-rm xdebug.ini
+# # Composer runs faster without Xdebug, and we don't need Xdebug any more
+# phpenv config-rm xdebug.ini
```
Comment (pauarge) [inline ci/generate-docs.sh:16]: Is there any reason to comment this line?

**Review by pauarge — CHANGES_REQUESTED** [review]
Amazing job! Left two small comments.

Also, remember that we need to unselect Travis as a required pipeline to run on a PR.

File: ci/deploy.sh (hunk 1 above)
Comment (pschoffer) [reply ci/deploy.sh]: Fixed.

File: ci/generate-docs.sh (hunk 2 above)
Comment (pschoffer) [reply ci/generate-docs.sh]: Just that the php docker image doesn't come with `phpenv` and it runs fast enough without this optimization so I felt, that it is simpler to just drop this. I did forget to remove the commented code though, it is removed now..

**Review by pauarge — APPROVED** [review]
LGTM

**Comment by pschoffer** [comment]
No production code change.

**Pull request merged**